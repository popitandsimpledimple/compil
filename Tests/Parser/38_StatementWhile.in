var a: integer = 0;
begin

while a < 10 do 
    begin
    a := 1 + a;
    writeln(a);
    end;

while ((a <> 6) and (a > 5)) do 
    begin
    a := 1 - a;
    writeln(a);
    end;

end.