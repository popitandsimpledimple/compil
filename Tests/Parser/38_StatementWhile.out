program
   decls
      vars
         var
            a
            type
               integer
               0
   block
      compounds statements
         while
            <
               a
               10
            compounds statements
               :=
                  a
                  +
                     1
                     a
               call
                  writeln
                     a
         while
            and
               <>
                  a
                  6
               >
                  a
                  5
            compounds statements
               :=
                  a
                  -
                     1
                     a
               call
                  writeln
                     a
